build_runner/lib/src/generate/build_impl.dart

```diff
@@ -335,6 +341,12 @@ class _SingleBuild {
       if (phase.isOptional) continue;
       await _performanceTracker.trackBuildPhase(phase, () async {
         if (phase is InBuildPhase) {
+          // Bail early if building a non-root package and we are building
```

Comment:
do we need to do something to convert a non-optional phase action into something that can be run lazily in this case?

Reply:
this is done btw, going to add a test for and check the logic for the optional outputs tracker as well.

Reply:
this is done as well, PTAL